code/datums/elements/atmos_requirements.dm

```diff
@@ -6,21 +6,38 @@
 /datum/element/atmos_requirements
 	element_flags = ELEMENT_BESPOKE
 	argument_hash_start_idx = 2
+	argument_hash_end_idx = 3
 	/// An assoc list of "what atmos does this mob require to survive in".
 	var/list/atmos_requirements
 	/// How much (brute) damage we take from being in unsuitable atmos.
 	var/unsuitable_atmos_damage
 
-/datum/element/atmos_requirements/Attach(datum/target, list/atmos_requirements, unsuitable_atmos_damage = 5)
+/datum/element/atmos_requirements/Attach(datum/target, list/atmos_requirements, unsuitable_atmos_damage = 5, mapload = FALSE)
 	. = ..()
 	if(!isliving(target))
 		return ELEMENT_INCOMPATIBLE
 	if(!atmos_requirements)
-		stack_trace("[type] added to [target] without any requirements specified.")
+		CRASH("[type] added to [target] without any requirements specified.")
+	var/list/short = atmos_requirements
+	if(!short["min_oxy"] && !short["max_oxy"] && !short["min_plas"] && !short["max_plas"] && !short["min_co2"] && !short["max_co2"] && !short["min_n2"] && !short["max_n2"])
+		CRASH("[type] added to [target] with all requirements set to 0.")
 	src.atmos_requirements = atmos_requirements
 	src.unsuitable_atmos_damage = unsuitable_atmos_damage
 	RegisterSignal(target, COMSIG_LIVING_HANDLE_BREATHING, PROC_REF(on_non_stasis_life))
 
+	if(!mapload || !PERFORM_ALL_TESTS(focus_only/atmos_and_temp_requirements) || is_breathable_atmos(target))
+		return
+	var/mob/living/living_mob = target
+	if(living_mob.stat == DEAD)
+		return
+	var/turf/open/open_turf = living_mob.loc
+	var/list/gases
+	var/string_text = "No Air"
+	if(open_turf.air)
+		gases = get_atmos_req_list(open_turf)
+		string_text = "O2: [gases["o2"]] - Plasma: [gases["plasma"]] - N2: [gases["n2"]] - CO2: [gases["co2"]]"
+	stack_trace("[target] loaded on a turf with unsafe atmos at \[[open_turf.x], [open_turf.y], [open_turf.z]\] (area : [open_turf.loc]). Turf gases: [string_text]. Check the mob atmos requirements again.")
```

Comment:
can you spin all of this out into a separate proc on the element? feels kinda weird to have `Attach()` be so bulky when this is only done in a unit testing environment, could cause confusion issues to someone newer to the codebase.